src/components/LanguageSelector.tsx

```diff
@@ -0,0 +1,55 @@
+import { useState } from "react";
+import { Button, Menu } from "@mantine/core";
+import { IconLanguage } from "@tabler/icons-react";
+import { useTranslation } from "react-i18next";
+import data from '../i18n/locales/LanguageSelectorList.json'
+
+export function LanguageSelector() {
+  const {i18n} = useTranslation();
+
+  function getMatchI18nLanguage() {
+    const filter = data
+      .filter(item => item.locale.match(i18n.language));
+    return (filter.length > 0) ? filter.at(0) : undefined;
+  }
+
+  const matchI18nLanguage = getMatchI18nLanguage();
+
+  const [opened, setOpened] = useState(false);
+  const [selected, setSelected] = useState(matchI18nLanguage ? matchI18nLanguage : data[0]);
+
+  const items = data.map((item) => (
+    <Menu.Item
+      onClick={() =>
+        i18n.changeLanguage(item.locale)
+          .then(() => setSelected(item))}
+      key={item.locale}
+    >
+      {item.variant}
+    </Menu.Item>
+  ));
+
+  return (
+    <Menu
+      onOpen={() => setOpened(true)}
```

Comment:
I don't believe you need a controlled menu here, do you ?

Reply:
I'm new to mantine, but I thought it was a good way to get it to manage selection. I'll have a look at other components to see if there's anything more stuitable

Reply:
btw, le me know if you have anything in mind :)

Reply:
Oh, I juste mean you can remove the open state and the onOpen on close props it will manage that auto